lib/pages/trade/desktop_trade.dart

```diff
@@ -50,668 +50,683 @@ class _DesktopTradeState extends State<DesktopTrade> {
     final tokenContainerWid = wid * 0.95;
     final amountBoxAndMaxButtonWid = tokenContainerWid * 0.5;
 
-    return BlocConsumer<TradePageBloc, TradePageState>(
-      listenWhen: (_, current) => current.status == BlocStatus.error,
-      listener: (context, state) {
-        if (state.failure is NoSwapInfoFailure) {
-          context.showWarningToast(
-            title: 'Action Error',
-            description: 'No swap info found',
-          );
+    return BlocListener<WalletBloc, WalletState>(    
+      listener: (_, state) {
+        if (state.isWalletConnected || state.isWalletDisconnected) {
+          context.read<TradePageBloc>().add(WatchAppDataChangesStarted());
```

Comment:
..These are the only two states we can have right? It sounds like in that case this should just always be done.